.github/workflows/reusable_monitoring.yml

```diff
@@ -0,0 +1,112 @@
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# NOTE: This GHA should not be run concurrently.
+
+name: Rekor Monitoring Template
+
+on:
+  workflow_call:
+    inputs:
+      file_issue:
+        description: 'True to file an issue on monitoring failure'
+        required: true
+        type: boolean
+
+permissions:
+  contents: read
+
+env:
+  UPLOADED_LOG_NAME: checkpoint
+  LOG_FILE: checkpoint_log.txt
+
+jobs:
+  detect-workflow:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write # Needed to detect the current reusable repository and ref.
+    outputs:
+      repository: ${{ steps.detect.outputs.repository }}
+      ref: ${{ steps.detect.outputs.ref }}
+    steps:
+      - name: Detect the repository and ref
+        id: detect
+        uses: slsa-framework/slsa-github-generator/.github/actions/detect-workflow@68bad40844440577b33778c9f29077a3388838e9 # v1.4.0
+
+  monitor:
+    runs-on: ubuntu-latest
+    needs: [detect-workflow]
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          repository: ${{ needs.detect-workflow.outputs.repository }}
+          ref: "${{ needs.detect-workflow.outputs.ref }}"
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: 1.19
+      - name: Download artifact
+        uses: dawidd6/action-download-artifact@bd10f381a96414ce2b13a11bfa89902ba7cea07f # v2.24.3
+        with:
+          name: ${{ env.UPLOADED_LOG_NAME }}
+        # Skip on first run since there will be no checkpoint
+        continue-on-error: true
+      - name: Log current checkpoints
+        run: cat ${{ env.LOG_FILE }}
+        # Skip on first run
+        continue-on-error: true
+      - run: go run ./cmd/mirroring --file ${{ env.LOG_FILE }} --once
+      - name: Upload checkpoint
+        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # v3.1.1
+        with:
+          name: ${{ env.UPLOADED_LOG_NAME }}
+          path: ${{ env.LOG_FILE }}
+          retention-days: 1
```

Comment:
Very minor question here: what happens if someone runs their monitor weekly? In that case there would be no consistency measured across because the checkpoints would disappear.

Do you want to document that it's intended to be run on a daily schedule (at least)?

It seems hairier to detect the callers schedule. You can via `github.event.schedule`, but I can imagine you'd need to do some parsing and so on on the cron and make a decision for what the minimum time should be

Reply:
Excellent point, I'm going to just provide an input to allow the caller to set retention and document this as a limitation. I've increased it to 14 days also - The artifacts are tiny, so I'm not worried about storage limits.

Reply:
perfect!